lib_eio/task_pool.mli

```diff
@@ -0,0 +1,13 @@
+type 'a task = unit -> 'a
+
+type t
+
+val create : sw:Switch.t -> max_domains:int -> Domain_manager.t -> t
```

Comment:
When I had a look at adding support for a task pool abstraction, I went with baking it directly into the `Domain_manager.t` capability. Users could then request a slice of that resource and get some amount back (I had in my mind that in the future you could add some user-specified way to dynamically scale this). This was in order to ensure we never overcommitted the "parallelism" resource. By allowing people to create multiple pools, I think that means they must ensure the sum of the `max_domains` they pass doesn't exceed a certain limit (presumably `Domain.recommended_domain_count ()`) ? Perhaps we don't want to be in charge of managing the parallel resource in this way ?

Reply:
Personally I don't think Eio should be making the final decision here, since there are cases where you want OS preemptive scheduling to kick in even for your CPU heavy tasks, regardless of whether they actually run in parallel or not.

But I also don't know how expensive overcommitted domains or idling domains are.